### PR TITLE
Fix anonymous function param definitions and add code actions for let expr functions

### PR DESCRIPTION
**Pull request by jmbockhorst**
Fixed anonymous function param definitions and was able to enable some existing tests. This fixed some inference bugs. 
Also added a code action to add the type annotation to functions in let expr. 

Inference bugs fixed:
- WebGl type renders as `shader` (like intellij)
- Use module alias if there is one
- Fix functions that used anonymous functions with record or tuple param deconstructors (like `tuple` in `elm-codec`)

All other reported bugs behave the same as intellij-elm, so I don't think we should worry about those edge cases for now.

### CODE REVIEW

File: test/definitionProviderTests/miscDefinition.test.ts [hunk 1]
```diff
@@ -52,49 +52,50 @@ foo =
 
   // LAMBDAS (ANONYMOUS FUNCTIONS)
 
-  xit(`test lambda parameter ref`, async () => {
+  it(`test lambda parameter ref`, async () => {
     const source = `
 f = \\x -> x
    --X  --^
 `;
+    console.log(source);
```
Comment (Razzeee) [inline test/definitionProviderTests/miscDefinition.test.ts:60]: Should we remove this log line?

File: test/definitionProviderTests/miscDefinition.test.ts [hunk 2]
```diff
@@ -52,49 +52,50 @@ foo =
...
+  it(`test lambda parameter destructured tuple ref`, async () => {
     const source = `
-f = \(x,y) -> x
+f = \\(x,y) -> x
     --X     --^
 `;
     await testBase.testDefinition(source);
   });
 
   xit(`test lambda parameter destructured with alias`, async () => {
```
Comment (Razzeee) [inline test/definitionProviderTests/miscDefinition.test.ts:96]: I guess this one isn't fixed yet?

File: src/util/treeUtils.ts [hunk 3]
```diff
@@ -2156,6 +2145,11 @@ export class TreeUtils {
         return;
       }
 
+      const asClause = TreeUtils.findFirstNamedChildOfType(
```
Comment (Razzeee) [inline src/util/treeUtils.ts:2148]: Only do this if it's needed. So after the next `return`

File: test/definitionProviderTests/miscDefinition.test.ts (hunk 2 above)
Comment (jmbockhorst) [reply test/definitionProviderTests/miscDefinition.test.ts]: Turns out it was fixed.
